include/pybind11/eigen/matrix.h

```diff
@@ -287,6 +287,8 @@ handle eigen_encapsulate(Type *src) {
 template <typename Type>
 struct type_caster<Type, enable_if_t<is_eigen_dense_plain<Type>::value>> {
     using Scalar = typename Type::Scalar;
+    static_assert(!std::is_pointer<Scalar>::value,
```

Comment:
I think you might be able to move these asserts to EigenProps, which would reduce the amount of redundant code.

Reply:
I was ambivalent before and still am a little bit, but decided to keep the 5 `static_assert()`:

Cons: 2-3 x 2 more lines of code.

Pros:

The `static_assert()`s appear at or near the top of each of the `type_caster`s, which makes them essentially a kind of documentation.

In case a `static_assert()` fails, it will be much more obvious that the message we want to send is: the `type_caster` does not support pointer types as scalar types.